### DISCUSSION TITLE
Sign-up database trigger to insert into public users table

### DISCUSSION THREAD

**Discussion by w2ca** [supabase/supabase#27154] [Questions]
I'm following the documentation to execute a trigger after a user signs up, the point is that it works well with signup credentials, however when I use Google to sign in it gives an error, this is because Google doesn't show me the given_name in the app_raw_user_metadata (first_name) and family_name(last_name).

What google gives me is this.

```json
{
 "iss": "https://accounts.google.com",
 "sub": "...",
 "name": "Jhon Doe",
 "email": "example@mail.com",
 "picture": "url",
 "full_name": "Jhon Doe",
 "avatar_url": "url",
 "provider_id": "...",
 "email_verified": true,
 "phone_verified": false
}
```

I would like there to be more data here, such as given_name and family_name.

I have already enabled the profile, email and openid scopes and I do not receive this data.

Can I request the given_name and family_name?

**Comment by GaryAustin1** [comment]
When you say enabled, did you add those scopes to the signInWithOAuth() call?

**Reply by w2ca** [reply]
Yes, I added it to the Google Cloud Console and to the request as well.

**Reply by GaryAustin1** [reply]
Is the scope data in identities of the user data?  
![image](https://github.com/supabase/supabase/assets/54564956/f857c0d9-723a-440b-b221-71338861a3e0)

The user metadata may only copy in some data and you mention a trigger.  A trigger on auth.users would not contain data going to auth.identities table.

I don't currently have OAuth setup so can't do a quick check.

**Reply by w2ca** [reply]
I ended up not finding a way to obtain the given_name and family_name during Google authentication, so I chose to change my strategy and started using the name. Thank you for your attention.
